### PR TITLE
Transformations: Add transformation builder tests and refactor

### PR DESCRIPTION
**Pull request by baldm0mma**
**What is this feature?**

This adds tests to the markdown building script for transformations, plus requisite tests

**Why do we need this feature?**

Tests are important, m'kay???

**Who is this feature for?**

Devs

**Which issue(s) does this PR fix?**:

<!--

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: scripts/docs/generate-transformations.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+export function contentDoesMatch(jsContent: string, mdContent: string): Boolean {
+  return normalizeContent(jsContent) === normalizeContent(mdContent);
+}
+
+/* 
+  Normalize content by removing all whitespace (spaces, tabs, newlines, carriage returns, 
+  form feeds, and vertical tabs) and special characters.
+
+  NOTE: There are numerous unpredictable formatting oddities when pasring javascript to markdown;
+  almost all of them are irrelevant to the actual content of the file, which is why we strip them out here.
```
Comment (jdbaldry) [inline scripts/docs/generate-transformations.test.ts:28]: One typo and some nits.

```suggestion
  NOTE: There are numerous unpredictable formatting oddities when transforming JavaScript to Markdown;
  almost all of them are irrelevant to the actual content of the file, which is why we strip them out here.
```

On the technicalities of this statement, I think most (probably all) Markdown parsers would correctly parse the JavaScript table as it is. I think it is a `prettier` formatting change that results in the Markdown table differences.

That being said, I think it's worth using the `prettier` formatted Markdown for a few reasons:
1. It's simpler for tools to roundtrip modifications. 
1. It helps alleviate the ambiguities in the Markdown specs.
1. Consistent formatting is easier for humans to read and parse.

I guess what we are actually talking about is parsing differences between `prettier` and whatever parsing library is used by the Grafana UI. I'm not sure how important that distinction is in terms of function and variable naming, just thought it's worth mentioning :)

File: scripts/docs/generate-transformations.test.ts
Comment (jdbaldry) [inline scripts/docs/generate-transformations.test.ts:1]: Does the presence of this test mean that now CI will fail if something breaks the Markdown generation?

Comment (baldm0mma) [reply scripts/docs/generate-transformations.test.ts:1]: That is correct. I wanted to make sure that anytime the content was changed, it was reflected in the markdown docs as well. I added some annotations to make sure that correcting the issue would be easy. Do you have concerns that I'm possibly not thinking about?

Comment (jdbaldry) [reply scripts/docs/generate-transformations.test.ts:1]: 🥳 ❤️

**Review by jdbaldry — APPROVED** [review]
Works for me. I'm not sure if you should consider my review definitive, I've only checked the result of rebuilding the documentation with Make and not interrogated the code itself. At a scan it looks reasonable :)

**Pull request merged**